data-prepper-plugins/rss-source/src/main/java/org/opensearch/dataprepper/plugins/source/rss/RssReaderTask.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.source.rss;
+
+import com.apptasticsoftware.rssreader.Item;
+import com.apptasticsoftware.rssreader.RssReader;
+import org.opensearch.dataprepper.model.buffer.Buffer;
+import org.opensearch.dataprepper.model.document.Document;
+import org.opensearch.dataprepper.model.document.JacksonDocument;
+import org.opensearch.dataprepper.model.record.Record;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+class RssReaderTask implements Runnable {
+
+    private static final Logger LOG = LoggerFactory.getLogger(RssReaderTask.class);
+
+    private final RssReader rssReader;
+
+    private final RSSSourceConfig rssSourceConfig;
+
+    private final Buffer<Record<Document>> buffer;
+
+    final Collection<Record<Document>> collection = new HashSet<>();
+
+
+    public RssReaderTask(RssReader rssReader, final RSSSourceConfig rssSourceConfig, final Buffer<Record<Document>> buffer) {
```

Comment:
It is probably good idea to just pass the URL here instead of rssSourceConfig as the argument?

Reply:
That would not be accurate. There's also `pollingFrequency` which by default will be 5 minutes. But it would be an optional argument the user can configure. I'll be adding it in a future PR.

Reply:
That may be true but I think, since this is an independent class, making it dependent on some thing like RSSSourceConfig is not ideal. Any necessary info should be explicitly passed instead of passing the entire config. I am not strict about this. Just a suggestion. What do you think @dlvenable

Reply:
I tend to agree with @kkondaka  here. I think the RSSSource can just pass in what RssReaderTask needs. But, I also don't feel strong about this.

Also, `pollingFrequency` should never be needed here since the scheduling happens in RSSSource.

One configuration that might come up is the write timeout.

Reply:
Understood, I decided to update this.